drivers/modem/modem_receiver.c

```diff
@@ -224,9 +224,9 @@ int mdm_receiver_register(struct mdm_receiver_context *ctx,
 		return -EINVAL;
 	}
 
-	ctx->uart_dev = device_get_binding(uart_dev_name);
+	ctx->uart_dev = uart_dev;
 	if (!ctx->uart_dev) {
-		LOG_ERR("Binding failure for uart: %s", uart_dev_name);
+		LOG_ERR("Binding failure for uart: %p", uart_dev);
 		return -ENODEV;
 	}
```

Comment:
shouldn't that be replaced with a call to `device_is_ready()`? (returns false if not ready or pointer is NULL)

Reply:
Possibly. I will get back to it once unit tests will pass, as I am not aware when and how should `device_is_ready()` be used yet.

Reply:
I've added `device_is_ready()` here and in `modem_iface_uart_init_dev()`. Thanks for hint!